### ISSUE TITLE
Investigate changes in Vim 9

### ISSUE THREAD

**Issue by petdance** [vim-perl/vim-perl#322]
Can we take advantage of any changes in Vim 9?

What about speedups using Vim9 script? https://vimhelp.org/version9.txt.html#new-9

> Compiled [functions](https://vimhelp.org/eval.txt.html#functions) [a](https://vimhelp.org/insert.txt.html#a)re introduced which allow for a large speed improvement.  You can expect around ten times faster execution, or even more.  The price to pay [is](https://vimhelp.org/motion.txt.html#is) that [Vim9](https://vimhelp.org/vim9.txt.html#Vim9) [script](https://vimhelp.org/usr_41.txt.html#script) [is](https://vimhelp.org/motion.txt.html#is) not backwards compatible.

**Comment by dkearns** [comment]
Yes, it would be worth converting the indent script to Vim9 script.

Neovim doesn't currently support Vim9 script.  There's some work to write a compiler but when or if that will be merged is unknown. 

We could keep a Legacy script version synced with the new Vim9 script version without too much effort.

**Comment by dkearns** [comment]
Vim9 script gets us a small performance improvement 20s vs 25s when indenting `perl5db.pl`.  The current implementation is spending all all of its time in expensive syntax ID tests.
